File: src/main/Java/boj_step20/boj2108.java
import java.io.*;
import java.util.Arrays;
import java.util.stream.IntStream;

public class boj2108 {
    public static void main(String[] args) throws IOException {
        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter writer = new BufferedWriter(new OutputStreamWriter(System.out));
        int n = Integer.parseInt(reader.readLine());
        int[] list = new int[n];
        int[] count = new int[8001];
        for (int i = 0; i < n; i++) {
            list[i] = Integer.parseInt(reader.readLine());
            count[list[i] + 4000]++;
        }
        Arrays.sort(list);
        int maxCount = Arrays.stream(count).max().getAsInt();
        int maxCountResult = IntStream.range(0, count.length).filter(i -> count[i] == maxCount).findFirst().getAsInt() - 4000;
        if (IntStream.range(0, count.length).filter(i -> count[i] == maxCount).count() > 1) {
            maxCountResult = IntStream.range(0, count.length).filter(i -> count[i] == maxCount).skip(1).findFirst().getAsInt() - 4000;
        }
        writer.append(String.valueOf(Math.round(Arrays.stream(list).average().getAsDouble()))).append("\n");
        writer.append(String.valueOf(list[n / 2])).append("\n");
        writer.append(String.valueOf(maxCountResult)).append("\n");
        writer.append(String.valueOf(list[n - 1] - list[0])).close();
    }
}

File: src/main/boj_step20/boj2108.kt
import java.io.BufferedReader
import java.io.BufferedWriter
import java.io.InputStreamReader
import java.io.OutputStreamWriter
import kotlin.math.roundToInt

fun main() {
    val r = BufferedReader(InputStreamReader(System.`in`))
    val w = BufferedWriter(OutputStreamWriter(System.out))
    val n = r.readLine().toInt()
    val list = mutableListOf<Int>()
    val count = IntArray(8001)
    repeat(n) { list.add(r.readLine().toInt()) }
    list.forEach { count[it + 4000]++ }
    if ( count.count { it == count.maxOrNull() } >= 2) count[count.indexOf(count.maxOrNull()!!)]--
    list.sort()
    w.appendLine("${(list.sum().toDouble() / list.size).roundToInt()}")
    w.appendLine("${list.elementAt(n / 2)}")
    w.appendLine("${count.indexOf(count.maxOrNull()!!) - 4000}")
    w.appendLine("${list.last() - list.first()}")
    w.close()
}